hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/JpaDistributionSetType.java

```diff
@@ -108,37 +110,74 @@ public void setDeleted(final boolean deleted) {
 
     @Override
     public Set<SoftwareModuleType> getMandatoryModuleTypes() {
+        if (elements == null) {
+            return Collections.emptySet();
+        }
+
         return elements.stream().filter(element -> element.isMandatory()).map(element -> element.getSmType())
                 .collect(Collectors.toSet());
     }
 
     @Override
     public Set<SoftwareModuleType> getOptionalModuleTypes() {
+        if (elements == null) {
+            return Collections.emptySet();
+        }
+
         return elements.stream().filter(element -> !element.isMandatory()).map(element -> element.getSmType())
                 .collect(Collectors.toSet());
     }
 
     @Override
     public boolean areModuleEntriesIdentical(final DistributionSetType dsType) {
+        if (!(dsType instanceof JpaDistributionSetType) || isOneModuleListEmpty(dsType)) {
+            return false;
+        } else if (areBothModuleListsEmpty(dsType)) {
+            return true;
+        }
+
         return new HashSet<DistributionSetTypeElement>(((JpaDistributionSetType) dsType).elements).equals(elements);
     }
 
+    private boolean isOneModuleListEmpty(final DistributionSetType dsType) {
```

Comment:
little bit confused about the method name, is not about if one list is empty, is about if one is empty and the other not, or if the other is empty and the one not :) Just the name and the complex if was little bit hard to read.

Before there was tested if the `elements` were equal is this the same check by checking if there are empty?
`HashSet<DistributionSetTypeElement>(((JpaDistributionSetType) dsType).elements).equals(elements);`